src/os/bluestore/simple_bitmap.h

```diff
@@ -0,0 +1,112 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*- 
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Author: Gabriel BenHanokh <benhanokh@gmail.com>
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software 
+ * Foundation.  See file COPYING.
+ * 
+ */
+#pragma once
+#include <cstdint>
+#include <iostream>
+#include <string>
+#include <vector>
+#include <unordered_map>
+#include <cstring>
+#include <cmath>
+#include <cassert>
+#include <iomanip>
+
+#include "include/ceph_assert.h"
+
+struct extent_t {
+  uint64_t offset;
+  uint64_t length;
+};
+
+class SimpleBitmap {
+public:
```

Comment:
SimpleBitmap is more complicated then it should.
There is no need for it to contain iteration state for finding free regions.

Imagine 2 simple, stateless functions:
`uint64_t find_0(uint64_t offset)`
and
`uint64_t find_1(uint64_t offset)`
that will find next 0 or 1 bit starting from offset.
Using those 2 it will be trivial to build function
`extent_t get_next_free(uint64_t offset)`
that will also be completely stateless.

Reply:
... and that functionality can be added on top of dynamic_bit...

Reply:
@ronen-fr 
Not really possible. boost::dynamic_bitset privatizes underlying container.